cmd/kubernetes.go

```diff
@@ -0,0 +1,818 @@
+		if err != nil {
+			die("Could not authenticate against the cluster: %s", err)
+		}
+
+		// Check if an existing deployment with the label 'app=wr-manager' exists.
+		// Read in namespace from resource file, if no file exists and no namespace
+		// is passed as a flag create new namespace and redeploy.
+		// If a resource file exists or a namespace is passed as a flag check to see if
+		// there is an existing manager deployment to reconnect to
+
+		var kubeDeploy bool //defaults false
+
+		// If a namespace is passed it takes priority.
+		if len(kubeNamespace) != 0 {
+			_, err = c.Clientset.AppsV1().Deployments(kubeNamespace).Get("wr-manager", metav1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					// set the flag that tells the daemon to redeploy.
+					kubeDeploy = true
+				} else {
+					die("wr-manager deployment found but not healthy: %s", err)
+				}
+			}
+		} else {
+			// Look for a set of resources in the manager directory
+			// If found, load them, otherwise use a new empty set.
+			if _, serr := os.Stat(resourcePath); os.IsNotExist(serr) {
+				kubeDeploy = true
+			} else {
+				// Read the namespace resource file
+				file, err := os.Open(resourcePath)
+				if err != nil {
+					die("could not open resource file with path: %s", err)
+				}
+				decoder := gob.NewDecoder(file)
+				err = decoder.Decode(resources)
+				if err != nil {
+					die("error decoding resource file: %s", err)
+				}
+
+				namespace := resources.Details["namespace"]
+				internal.LogClose(appLogger, file, "resource file", "path", resourcePath)
+
+				// check for a healthy deployment
+				_, err = c.Clientset.AppsV1().Deployments(namespace).Get("wr-manager", metav1.GetOptions{})
+				if err != nil {
+					if errors.IsNotFound(err) {
+						// set the flag that tells the daemon to redeploy.
+						kubeDeploy = true
+					} else {
+						die("wr-manager deployment found but not healthy: %s", err)
+					}
+				}
+			}
+		}
+
+		if !kubeDeploy {
+			info("found existing healthy wr-manager deployment, reconnecting")
+		} else {
+			info("please wait while wr is deployed to the cluster")
+		}
+
+		// Daemonise
+		fwPidPath := filepath.Join(config.ManagerDir, "kubernetes_resources.fw.pid")
+		umask := 007
+		child, context := daemonize(fwPidPath, umask, extraArgs...)
+		if child != nil {
+			// PostParent() (Runs in the parent process after spawning child)
+			jq = connect(120*time.Second, true)
+			if jq == nil {
+				die("could not talk to wr manager after 120s")
+			}
+
+			// The remote manager is running, read the resource file to determine the name
+			// of the pod to fetch the client.token from.
+
+			// Read the manager pod's name from resource file
+			file, err := os.Open(resourcePath)
+			if err != nil {
+				die("Could not open resource file with path: %s", err)
+			}
+			decoder := gob.NewDecoder(file)
+			err = decoder.Decode(resources)
+			if err != nil {
+				die("Error decoding resource file: %s", err)
+			}
```

Comment:
reading and decoding resourcePath is done above as well; turn in to a function that is called in both places